src/connectComp.cpp

```diff
@@ -0,0 +1,265 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <vector>
+#include <iostream>
+
+#include <unordered_map>
+
+#include <unordered_set>
+#include <string.h>
+#include <cassert>
+#include <cstdlib>
+
+namespace found {
+
+struct Point {
+	int x;
+	int y;
+};
+
+struct Edge {
+	int magnitude; // number of points in this edge
+	std::vector<Point> points;
+	// int Xmin, Xmax, Ymin, Ymax;
+};
+
+std::vector<Edge> NaiveConnectedComponent(unsigned char *image, int imageWidth, int imageHeight) { // const??
```

Comment:
You may go ahead and const the image field, but you should do this at both the pointer level and the value level. This will make sure that you cannot change neither the value nor the reference stored by this pointer. If its inconvenient, it does not need to be done though.